### ISSUE TITLE
Trajectories for projectile weapons

### ISSUE THREAD

**Issue by SylveonBottle** [Fedoraware/Fedoraware#832]
![image](https://user-images.githubusercontent.com/17802843/204065489-5d3c2c03-786b-431f-a70c-c68a41b1856e.png)

basically this but for TF2 please. 

-The "predicted trajectory" line take account of charing sticky/bow
-Show Trajectory line for enemies that CAN fire (so we know that we may be in danger)
-Show Trajectory line on projectiles ALREADY fired (so we know that we ARE in danger, and can dodge)

![image](https://user-images.githubusercontent.com/17802843/204064329-bca31577-4e91-451f-928e-705b3186953b.png)

![image](https://user-images.githubusercontent.com/17802843/204064721-de7a0064-4698-4fca-8949-b8997b48b7e7.png)

Although the pill is inaccurate, we can use it as a basis for if our shot will make it over an obstacle or not.
![image](https://user-images.githubusercontent.com/17802843/204065589-949338c2-2cb8-4a31-b86e-9a90e946d6b0.png)
![image](https://user-images.githubusercontent.com/17802843/204065702-6a4b2631-f7fc-4079-b4d5-b3cd7ae52340.png)

Animated examples of charging:
![image](https://user-images.githubusercontent.com/17802843/204065514-3e61ab34-46d7-4738-83af-05a2df0de2a3.gif)

(crappy) Examples of Trajectories on enemies:
![image](https://user-images.githubusercontent.com/17802843/204065954-f02f2aa9-272c-478c-9eb8-0d334f69f0a5.png)
Thanks to Trajectories, we can tell that the flare is going to our left, and we know to dodge to our right.
Hasn't fired yet:
![image](https://user-images.githubusercontent.com/17802843/204066154-7e0fba9e-44cb-4d6f-89e6-1ada45916809.png)
Tracking projectile (pyro is strafing to his left and is reloading his next shot, we don't need to show a line coming from him anymore)
![image](https://user-images.githubusercontent.com/17802843/204066226-a71d00a1-d522-4e53-9667-8b0989ecbb4c.png)

![image](https://user-images.githubusercontent.com/17802843/204066511-9ef490da-db55-48fb-bdcf-49408e03ee0b.png)

**Comment by SylveonBottle** [comment]
Yes I know demoman's pills are inaccurate.
Still would be useful to have a general idea of where it's going.
Also, being able to tell where loose cannon will explode before letting go of mouse1 would be OP

**Comment by frel0** [comment]
do it urself ezz retard

**Comment by SylveonBottle** [comment]
> do it urself ezz retard

shit why didn't I think of that 😓 
give me 4-6 years I should know how to code in C++ by then

**Comment by johnathon-walnut** [comment]
Honestly an easier compromise would be to just add the sniper sightlines to everyone all the time - that way you can see if someones gaze is roughly near you

**Comment by SylveonBottle** [comment]
> Honestly an easier compromise would be to just add the sniper sightlines to everyone all the time - that way you can see if someones gaze is roughly near you

Yeah but this gets super annoying seeing random lines everywhere.
How about only showing lines for players that are also roughly looking in your direction? 
And the main focus of this suggestion is just being able to see where _your_ gravity-effected projectiles will likely land.

**Comment by Podssa** [comment]
sounds like premium hack fetures

**Comment by vaitomanocucpp** [comment]
that would be actually helpfull

**Comment by stax132349** [comment]
https://streamable.com/ojrfj7 :trollface:

**Comment by SylveonBottle** [comment]
> https://streamable.com/ojrfj7 :trollface:

nice for initial trajectories, but I was thinking more like recalculating the trajectories every tick or so, using the current projectile's position+velocity as a base.
That way if it gets effected by anything (like pyro airblast) it can still show properly. And also this method would remove the line behind the arrow, which you can see clutters hardcore in your vid

**Comment by stax132349** [comment]
https://streamable.com/luld43 this isnt made by me btw

**Comment by SylveonBottle** [comment]
> https://streamable.com/luld43 this isnt made by me btw

Trajectory lines updated by aimbot offset? That's actually pretty awesome. 



> https://streamable.com/ibz9f8 that line behind the arrow thing got fixed as well earlier

That doesn't look fixed.
I mean, the actual line _behind_ the arrow should disappear. As in, the line shouldn't.
